### PR TITLE
Improving AWS cluster creation docs

### PR DESCRIPTION
**Pull request by sgibson91**
This PR improves the documentation for deploying a new AWS cluster.

### CODE REVIEW

File: docs/howto/operate/new-cluster/aws.md [hunk 1]
```diff
@@ -101,6 +105,8 @@ you at least one core node running.
 
 ### Grant access to other users
 
+<!-- Is this section still relevant if the account is managed by SSO? -->
+
```
Comment (sgibson91) [inline docs/howto/operate/new-cluster/aws.md:109]: @damianavila @yuvipanda Your input would be appreciated here

File: docs/howto/operate/new-cluster/aws.md [hunk 2]
```diff
@@ -128,6 +134,13 @@ role - and users can modify them as they wish.
 
 ### Create account with finely scoped permissions for automatic deployment
 
+<!-- Running terraform for AWS code now does more than *just* create a CI/CD account -
```
Comment (sgibson91) [inline docs/howto/operate/new-cluster/aws.md:137]: I think?

File: docs/howto/operate/new-cluster/aws.md [hunk 3]
```diff
@@ -128,6 +134,13 @@ role - and users can modify them as they wish.
...
+we should update this section to reflect that.
+
+Also, the .tfvars file generated by `deployer generate-cluster` for aws is not complete.
+db_instance_identifier is required even if db_enabled is set to false. Is it possible
+to set this to something reasonable so that tf plan passes? -->
```
Comment (sgibson91) [inline docs/howto/operate/new-cluster/aws.md:142]: @yuvipanda your input would be appreciated here

Comment (yuvipanda) [reply docs/howto/operate/new-cluster/aws.md:146]: @sgibson91 I've made `db_instance_identifier` not be required now!

File: docs/howto/operate/new-cluster/aws.md (hunk 1 above)
Comment (yuvipanda) [reply docs/howto/operate/new-cluster/aws.md:113]: @sgibson91 unfortunately yes! However, the need for this is a lot lesser now thanks to your `use-cluster-credentials`!

File: docs/howto/operate/new-cluster/aws.md (hunk 2 above)
Comment (yuvipanda) [reply docs/howto/operate/new-cluster/aws.md:141]: It does:

1. Create CI/CD role 
2. Setup EFS in appropriate networked place
3. Optionally set up a shared database
4. Optionally set up user buckets

I think these are all documented but scattered

Comment (sgibson91) [reply docs/howto/operate/new-cluster/aws.md:141]: Ok, I will track these docs down and link them. And restructure this section a little bit to make more sense.

**Review by GeorgianaElena — APPROVED** [review]
Thanks for improving these @sgibson91!

**Pull request merged**